src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -550,27 +568,58 @@ vector<SparqlTripleSimple> Visitor::visit(Parser::QuadDataContext* ctx) {
     checkAndReportVar(quad.s_);
     checkAndReportVar(quad.p_);
     checkAndReportVar(quad.o_);
+    if (std::holds_alternative<Variable>(quad.g_)) {
+      reportError(ctx->quads(), "Variables are not allowed as graph names.");
+    }
   }
 
   return quads;
 }
 
+vector<SparqlTripleSimpleWithGraph> Visitor::transformTriplesTemplate(
+    Parser::TriplesTemplateContext* ctx,
+    const std::variant<Iri, Variable, std::monostate>& graph) {
+  auto convertTriple = [&graph](const std::array<GraphTerm, 3>& triple)
+      -> SparqlTripleSimpleWithGraph {
+    return {visitGraphTerm(triple[0]), visitGraphTerm(triple[1]),
+            visitGraphTerm(triple[2]), graph};
+  };
+
+  return ad_utility::transform(visit(ctx), convertTriple);
+}
 // ____________________________________________________________________________________
-vector<SparqlTripleSimple> Visitor::visit(Parser::QuadsContext* ctx) {
-  if (!ctx->quadsNotTriples().empty()) {
-    // Could also be default; disallow completely for now.
-    reportNotSupported(ctx->quadsNotTriples(0), "Named graphs are");
-  }
+vector<SparqlTripleSimpleWithGraph> Visitor::visit(Parser::QuadsContext* ctx) {
+  auto triplesWithGraph = ad_utility::transform(
+      ctx->triplesTemplate(), [this](Parser::TriplesTemplateContext* ctx) {
+        return transformTriplesTemplate(ctx, std::monostate{});
+      });
+  std::ranges::move(visitVector(ctx->quadsNotTriples()),
+                    std::back_inserter(triplesWithGraph));
+  return ad_utility::flatten(std::move(triplesWithGraph));
```

Comment:
You are reordering the triples here. Is that correct, because the order has no semantics? Then write a short comment to make it clearer.